pkg/operator/controller/gateway-service-dns/controller.go

```diff
@@ -241,8 +242,28 @@ func (r *reconciler) ensureDNSRecordsForGateway(ctx context.Context, gateway *ga
 		Name:       service.Name,
 		UID:        service.UID,
 	}
+	ics := operatorv1.IngressControllerList{}
+	if err := r.cache.List(ctx, &ics); err != nil {
+		log.Error(err, "failed to list Ingress Controllers")
+		return []error{err}
+	}
 	var errs []error
 	for _, domain := range domains {
+		// Check if the domain matches one from an existing ingress controller. If so, log an error and ignore the invalid domain.
+		domainClashes := false
```

Comment:
You don't really need this boolean, just break from the loop when you see the match.

Reply:
As written, this boolean is needed because you can't `continue` or `break` out of both the inner and outer `for` loop from inside the inner one. However, I reorganized the new code here to not need the boolean and I think it's more readable the new way.